molecule/verifier/lint/precommit.py

```diff
@@ -0,0 +1,146 @@
+#  Copyright (c) 2019 Red Hat, Inc.
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to
+#  deal in the Software without restriction, including without limitation the
+#  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+#  sell copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#  DEALINGS IN THE SOFTWARE.
+"""Linter module for pre-commit related code."""
+
+import os
+
+import sh
+
+from molecule import logger
+from molecule import util
+from molecule.verifier.lint import base
+
+LOG = logger.get_logger(__name__)
+
+
+class PreCommit(base.Base):
+    """
+    Pre-commit tool verifier wrapper.
```

Comment:
I think it should be clearer that this linter only checks files called `test_*.py` in the verifier directory.

Reply:
I thought that was implicit by the placement of the class. Shouldn't every option for a verifier lint command only operate on the files in the `tests` directory of the scenario?

Reply:
I guess I'm just not seeing an example, the other verifier lint commands don't have any mention of the files they operate on in their documentation. So I would recommend fixing the documentation for all of them at once if this is important to resolve.

Reply:
Not everyone reads the class definitions. A "this tool lints against files in the test folder with a prefix of ``test_``" would do it. There are other verifiers which are "incomptabile" (don't get any linting benefit) with this linter (like the Ansible verifier).

Reply:
Hopefully that documentation will work for you.